### DISCUSSION TITLE
Preload === Joins ? [Lucid]

### DISCUSSION THREAD

**Discussion by samyan** [adonisjs/core#4623] [Help]
Hello. I dont find information about joins though Models. Looking in docs, I understand that Preload is separeted query. Can anyone confirm it pls?

Thanks in advance.

**Comment by thetutlage** [comment]
You can self write join using the query builder. https://lucid.adonisjs.com/docs/select-query-builder#join

**Reply by samyan** [reply]
I know, but it is already different, because the output is not with the desired structure as in models (nested models, etc.), It is a RAW result. It would be good to implement it in models...

**Reply by RomainLanz** [reply]
You can check the following AdonisJS 5 package: https://github.com/chirgjin/adonisjs-select-related
